src/components/Party.tsx

```diff
@@ -37,10 +37,10 @@ export const Party = ({ username, roomId }: PartyProps) => {
         className="mt-2 flex flex-col gap-4"
         onSubmit={(event) => {
           const form = new FormData(event.target as HTMLFormElement);
-          const text = form.get("message").valueOf();
-
+          const text = form.get("message")?.valueOf();
           socket.send(JSON.stringify({ user: username, text }));
           event.preventDefault();
+          (event.target as HTMLFormElement).reset();
```

Comment:
refactor (blocking) use the form variable defined above in the function.

```suggestion
          form.reset();
```

Reply:
Thanks, Nick! I've tried doing it but I'm getting a TS error: `Property 'reset' does not exist on type 'FormData'. Did you mean 'set'?ts(2551)`. 
According to chatGPT, this error is because the reset method is not a part of the FormData interface in TypeScript. The reset method is associated with the HTMLFormElement interface. 
That's why we need to cast `eventTarget` as `HTMLFormElement` so TS knows that it's a form element and we can use `reset()`.

Reply:
Oops! You're right! I'm on mobile. 😅 That's the form data, not the form. You could create a form variable before the form data and use it for the form data and then for the reset.

Reply:
😅
do you mean like this: 
```
onSubmit={(event) => {
          const formElement = event.target as HTMLFormElement;
          const form = new FormData(formElement);
          const text = form.get("message")?.valueOf();
          socket.send(JSON.stringify({ user: username, text }));
          event.preventDefault();
          formElement.reset();
        }}
```

or we could do this:

```
onSubmit={(event) => {
  const formElement = event.target as HTMLFormElement;
  const text = formElement.message.value; 
  socket.send(JSON.stringify({ user: username, text }));
  event.preventDefault();

  formElement.reset();
}};
```

Reply:
Either works. I'd just put the event.preventDefault() at the end, but it's not blocking getting this PR merged.

Reply:
I just realized that we are not using `event.preventDefault()` at the beginning of the form submission handler. I believe I've always used it as my first line to prevent the default form submission behavior. why do we prefer using it at the end in this case?

Reply:
It can be at the beginning or the end. The reason for the end is for progressive enhancement.

Say there are JavaScript errors, the form won't be able to submit, but it doesn't really apply in our case because we don't have a form submission being sent back doing a full form , we're just sending it over websockets (partykit).

[Austin Gil explains this well in the live stream we did](https://youtu.be/cIkQ_4CEUlA?t=2983).

Reply:
Makes sense! Thanks, Nick! Will def watch the video.

Reply:
This video deserved a subscribe from me! :D I'm your 423rd subscriber, Nick! 🎉 will leave it here for records haha 
<img width="355" alt="Screenshot 2024-01-23 at 8 21 34 PM" src="https://github.com/nickytonline/astro-partykit-starter/assets/25575320/8c05f196-5978-48f0-b237-3b5023ae4174">